src/neuroagent/scripts/avalidate_tool_calls.py

```diff
@@ -0,0 +1,251 @@
+"""Run validation on tool calls."""
+
+import argparse
+import asyncio
+import json
+import logging
+from typing import Any, Dict
+
+import aiohttp
+import pandas as pd
+
+logging.basicConfig(
+    level=logging.INFO,  # Set the logging level
+    format="%(asctime)s - %(levelname)s - %(message)s",  # Define the log message format
+)
+
+
+async def fetch_tool_call(
+    session: aiohttp.ClientSession,
+    query: Dict[str, Any],
+    base_url: str,
+    semaphore: asyncio.Semaphore,
+) -> Dict[str, Any]:
+    """
+    Fetch the tool call results for a given test case.
+
+    This function sends an asynchronous POST request to the API with the provided
+    test case data and validates the tool calls against expected outcomes.
+
+    Args:
+    ----
+    session (aiohttp.ClientSession): The aiohttp session used to make the HTTP request.
+    query (dict): A dictionary containing the test case data, including the prompt,
+                      expected tools, optional tools, and forbidden tools.
+    base_url (str): The base URL of the API.
+
+    Returns
+    -------
+    dict: A dictionary containing the prompt, actual tool calls, expected tool calls,
+          and whether the actual calls match the expected ones.
+    """
+    async with semaphore:
+        prompt = query["prompt"]
+        expected_tool_calls = query["expected_tools"]
+        optional_tools = query["optional_tools"]
+        forbidden_tools = query["forbidden_tools"]
+
+        logging.info(f"Testing prompt: {prompt}")
+
+        # Send a request to the API
+        async with session.post(
```

Comment:
This doesn't need to be a context manager.

Reply:
I resaerched the pros and cons of this vs await approach. Mainly , this one is better for resource management while await is better for flexibility.